geopandas/plotting.py

```diff
@@ -95,27 +118,17 @@ def plot_polygon_collection(
     # PatchCollection does not accept some kwargs.
     if "markersize" in kwargs:
         del kwargs["markersize"]
-    if color is not None:
-        if is_color_like(color):
-            kwargs["color"] = color
-        elif pd.api.types.is_list_like(color):
-            kwargs["color"] = np.take(color, multiindex, axis=0)
-        else:
-            raise TypeError(
-                "Color attribute has to be a single color or sequence of colors."
-            )
 
+    if color is not None:
+        kwargs["color"] = color
+        _check_and_expand(kwargs, "color", multiindex, is_color_like)
     else:
         for att in ["facecolor", "edgecolor"]:
             if att in kwargs:
-                if not is_color_like(kwargs[att]):
```

Comment:
Here is important to check `not is_color_like` before checking `is_list_like`, because colors can be lists. See #1198 and #1178.

Reply:
Thanks for the review! Are you sure this is incorrect? The `is_color_like` function is passed onto `_check_and_expand`, and there it's used to check for colorlikeness before doing testing if it's a list.

Reply:
In `_check_and_expand` is check if it **is** color_like. The original condition check if is **not** color_like. That is the difference I meant.